.dotnet/README.md

```diff
@@ -795,6 +796,52 @@ string message = outputAsJson.RootElement
 
 Notice how you can then call the resulting `ClientResult`'s `GetRawResponse` method and retrieve the response body as `BinaryData` via the `PipelineResponse`'s `Content` property.
 
+### Mock a client for testing
+
+The OpenAI .NET library has been designed to support mocking, providing key features such as:
+- Client methods made virtual to allow overriding.
+- Model Factories to assist in instantiating models that lack public constructors.
+
+To illustrate how mocking works, suppose you want to validate the behavior of the following method using the [Moq](https://github.com/devlooped/moq) library. Given the path to an audio file, it classifies whether it's a safe audio or not based on the presence of the substring "silly":
+
+```csharp
+public bool IsAudioUnsafe(AudioClient client, string audioFilePath)
+{
+    AudioTranscription transcription = client.TranscribeAudio(audioFilePath);
+    return transcription.Text.Contains("silly");
+}
+```
+
+Create mocks of `AudioClient` and `ClientResult<AudioTranscription>`, set up methods and properties that will be invoked, then test the behavior of the `IsAudioUnsafe` method. Since the `AudioTranscription` class does not provide public constructors, it must be instantiated by the `OpenAIAudioModelFactory` static class:
+
+```csharp
+// Instantiate mocks and the AudioTranscription object.
+
+Mock<AudioClient> mockClient = new();
+Mock<ClientResult<AudioTranscription>> mockResult = new(null, Mock.Of<PipelineResponse>());
+AudioTranscription transcription = OpenAIAudioModelFactory.AudioTranscription(text: "This is a silly transcription.");
```

Comment:
This is a clever example! But I wonder if it might confuse readers into thinking that it is related to the Moderations API. Maybe just to be safe we should change it? 🤔 

Here's an idea based on what you have:

```csharp
public bool ContainsSecretWord(AudioClient client, string audioFilePath, string secretWord)
{
    AudioTranscription transcription = client.TranscribeAudio(audioFilePath);
    return transcription.Text.Contains(secretWord);
}
```

And then you can pass some unamusing secret word like "apples". 😄